engine/access/ingestion/tx_error_messages/requester.go

```diff
@@ -0,0 +1,168 @@
+package tx_error_messages
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	execproto "github.com/onflow/flow/protobuf/go/flow/execution"
+	"github.com/rs/zerolog"
+	"github.com/sethvargo/go-retry"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/onflow/flow-go/engine/common/rpc/convert"
+	"github.com/onflow/flow-go/utils/logging"
+
+	"github.com/onflow/flow-go/engine/access/rpc/backend"
+	"github.com/onflow/flow-go/engine/common/rpc"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// RequesterConfig contains the retry settings for the tx error messages fetch.
+type RequesterConfig struct {
+	// the initial delay used in the exponential backoff for failed tx error messages download
+	// retries.
+	RetryDelay time.Duration
+	// the max delay used in the exponential backoff for failed tx error messages download.
+	MaxRetryDelay time.Duration
+}
+
+type Requester struct {
+	logger                     zerolog.Logger
+	config                     *RequesterConfig
+	backend                    *backend.Backend
+	execNodeIdentitiesProvider *rpc.ExecutionNodeIdentitiesProvider
+	executionResult            *flow.ExecutionResult
+}
+
+func NewRequester(
+	logger zerolog.Logger,
+	config *RequesterConfig,
+	backend *backend.Backend,
+	execNodeIdentitiesProvider *rpc.ExecutionNodeIdentitiesProvider,
+	executionResult *flow.ExecutionResult,
+) *Requester {
+	return &Requester{
+		logger:                     logger,
+		config:                     config,
+		backend:                    backend,
+		execNodeIdentitiesProvider: execNodeIdentitiesProvider,
+		executionResult:            executionResult,
+	}
+}
+
+// Request fetches transaction error messages for the specific
+// execution result this requester was configured with.
+//
+// No errors are expected during normal operations.
```

Comment:
This can actually return an error in 2 cases: context canceled and deadline exceeded. currently, the returned error would be a status error which I don't think is ideal. maybe do something like
```golang
	if err != nil {
		// allow unwrapping to underlying context error
		if ctx.Err() != nil {
			return errors.Join(err, ctx.Err())
		}
		return nil, err
	}
	return errorMessages, nil
```

then you could document those 2 error types here.